static/js/services/message-contacts.js

```diff
@@ -12,7 +12,7 @@ var async = require('async'),
     function(HttpWrapper, BaseUrlService) {
       return function(params, callback) {
         var url = BaseUrlService() + '/message_contacts';
-        HttpWrapper.get(url, { params: params })
+        HttpWrapper.get(url, { params: params, timeout: false })
```

Comment:
Because the Changes listener in MessagesCtrl passes in the `changes` option we can tell in updateContacts that we could pass in { timeout: false } to MessageContact. Then we'd have to check in the callback that no navigation had occurred.